src/UserSpaceInstrumentation/AccessTraceesMemory.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef USER_SPACE_INSTRUMENTATION_ACCESS_TRACEES_MEMORY_H_
+#define USER_SPACE_INSTRUMENTATION_ACCESS_TRACEES_MEMORY_H_
+
+#include <sys/types.h>
+
+#include <cstdint>
+#include <vector>
+
+#include "OrbitBase/Result.h"
+
+namespace orbit_user_space_instrumentation {
+
+// Read `length` bytes from process `pid` starting at `address_start`. `length` will be rounded up
+// to a multiple of eight. The data is returned in `bytes`.
+// Assumes we are already attached to the tracee `pid` e.g. using `AttachAndStopProcess`.
+[[nodiscard]] ErrorMessageOr<void> ReadTraceesMemory(pid_t pid, uint64_t address_start,
+                                                     uint64_t length, std::vector<uint8_t>* bytes);
```

Comment:
I think that this function should return `ErrorMessageOr<std::vector<uint8_t>>` instead of using an output parameter.

Reply:
done